webpack/scenes/Subscriptions/UpstreamSubscriptions/UpstreamSubscriptionsPage.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { LinkContainer } from 'react-router-bootstrap';
+import { Grid, Row, Col } from 'react-bootstrap';
+import { Button, Spinner } from 'patternfly-react';
+import Table from 'foremanReact/components/common/table';
+import PaginationRow from '../../../components/PaginationRow/index';
+import { columns } from './UpstreamSubscriptionsTableSchema';
+
+class UpstreamSubscriptionsPage extends Component {
+  componentDidMount() {
+    this.loadData();
+  }
+
+  loadData() {
+    this.props.loadUpstreamSubscriptions();
+  }
+
+  render() {
+    const { upstreamSubscriptions } = this.props;
+
+    const onPaginationChange = (pagination) => {
+      this.props.loadUpstreamSubscriptions({
+        ...pagination,
+      });
+    };
+
+    const emptyStateData = {
+      header: __('There are no Subscription Allocations to display'),
+      description: __('Subscription Allocations allow you to export subscriptions from the Red Hat Customer Portal to ' +
+          'an on-premise subscription management application such as Red Hat Satellite.'),
+      docUrl: 'http://redhat.com',
+      documentation: { title: __('Learn more about Subscription Allocations'), url: 'http://redhat.com' },
+      action: { title: __('New Subscription Allocation'), url: 'http://redhat.com' },
+    };
+
+    return (
+      <Grid bsClass="container-fluid">
+        <h1>{__('Add Subscriptions')}</h1>
+
+        <Row>
+          <Col sm={12}>
+            <Spinner loading={upstreamSubscriptions.loading} className="small-spacer">
+              <Table
+                rows={upstreamSubscriptions.results}
+                columns={columns}
+                emptyState={{ data: emptyStateData }}
+              />
+              <PaginationRow
+                viewType="table"
+                itemCount={upstreamSubscriptions.pagination.total}
+                pagination={upstreamSubscriptions.pagination}
+                onChange={onPaginationChange}
+              />
```

Comment:
I actually prefer this because it makes it clear where the component ends.  I think of it as an ending `}` or `)`.  If you feel strongly about it I can change.

Reply:
Your formatting is ok, I meant the margin as and "per page" as was mentioned in https://github.com/Katello/katello/pull/7230/files#r177057313 I'm sorry for the confusing comment :)
It's fixed now.